### ISSUE TITLE
Documentation for public APIs

### ISSUE THREAD

**Issue by mzabaluev** [movementlabsxyz/movement#81]
This is a meta-issue tracking code documentation for all public APIs in movement SDK.

Generally, any public items should have documentation. Mechanically, this is achievable by compiling with `#![deny(missing_docs)]` and can be backed by a CI check. Another level in quality is to have a human with technical writing experience review the generated documentation and make sure it's understandable and consistent.

**Comment by l-monninger** [comment]
Yeah, let's try to get the `#![deny(missing_docs)]` soon. We might be able to adopt a policy of "if you work on a crate, add this" to incrementally introduce.
